chia/wallet/wallet_state_manager.py

```diff
@@ -281,7 +285,7 @@ async def create_more_puzzle_hashes(
         targets = list(self.wallets.keys())
         self.log.debug("Target wallets to generate puzzle hashes for: %s", repr(targets))
         unused: Optional[uint32] = (
-            up_to_index if up_to_index is not None else await self.puzzle_store.get_unused_derivation_path()
+            uint32(up_to_index + 1) if up_to_index is not None else await self.puzzle_store.get_unused_derivation_path()
```

Comment:
The +1 is to account for range() not including the stop value (`last_index` in this case)